pkg/kubelet/eviction/helpers.go

```diff
@@ -172,6 +177,181 @@ func addAllocatableThresholds(thresholds []evictionapi.Threshold) []evictionapi.
 	return append(append([]evictionapi.Threshold{}, thresholds...), additionalThresholds...)
 }
 
+// UpdateContainerFsThresholds will add containerfs eviction hard/soft
+// settings based on container runtime settings.
+// Thresholds are parsed from evictionHard and evictionSoft limits so we will override.
+// If there is a single filesystem, then containerfs settings are same as nodefs.
+// If there is a separate image filesystem for both containers and images then containerfs settings are same as imagefs.
+func UpdateContainerFsThresholds(thresholds []evictionapi.Threshold, imageFs, separateContainerImageFs bool) ([]evictionapi.Threshold, error) {
```

Comment:
this makes sense.  thank you for the detailed comment.

can you confirm the effective thresholds (after detecting state of host configuration) written to a log for debuggability across all resource types (memory, disk, etc.)?

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/eviction/helpers.go#L772 logs all observations/thresholds if you run kubelet with more verbose logging.  

I use this to confirm that the separating disks works as we expect.